lib/ansible/plugins/lookup/template.py

```diff
@@ -98,12 +119,20 @@ def run(self, terms, variables, **kwargs):
                 vars.update(generate_ansible_template_vars(lookupfile))
                 vars.update(lookup_template_vars)
 
-                # do the templating
-                with self._templar.set_temporary_context(variable_start_string=variable_start_string,
-                                                         variable_end_string=variable_end_string,
-                                                         available_variables=vars, searchpath=searchpath):
-                    res = self._templar.template(template_data, preserve_trailing_newlines=True,
-                                                 convert_data=convert_data_p, escape_backslashes=False)
+                with templar.set_temporary_context(variable_start_string=variable_start_string,
+                                                   variable_end_string=variable_end_string,
+                                                   available_variables=vars, searchpath=searchpath):
+                    if USE_JINJA2_NATIVE:
+                        res = templar.do_template(template_data, preserve_trailing_newlines=True,
```

Comment:
Can we add a comment about why we call `.do_template` for native instead of `.template`?

Reply:
Looking at this again (because of course I forgot why I did that), the reason is effectively to bypass `Templar.is_possibly_template` call in `Templar.template`. If we had `'{\n      "key": "bar"\n}\n'` as an input for `is_possibly_template`, that would not be considered as a template and the original string would be returned. With native jinja2 we probably want it templated so it's turned into a `dict`. So I think what we should do here instead is to use `.template` for native and non-native in the template lookup and add (yet another) `if USE_JINJA2_NATIVE` check to `is_possibly_template` to always return `True` for native.

Sigh...

Reply:
So this turned out not to be a good idea as it broke the below task. Since `set_fact` is templated once and the result is stored (as opposed to a `vars:`), the below could be re-templated causing string (there are double quotes) being turned into integer.

```yaml
- set_fact:
    foo: "'{{ integer_bar }}'"
```

So it created more problems and calling `.template` for both native and non-native is what I ended up with.